File: Ex-25/ex_25_java/Calculadora.java
public class Calculadora {
    // Composição: A Calculadora possui (é dona de) Dados, Interface e Controle
    private CalcDados dados;
    private CalcInterface interface_;
    private CalcControle controle;

    // cria os componentes e os vínculos (via lista de inicialização)
    public Calculadora() {
        this.dados = new CalcDados();
        this.interface_ = new CalcInterface();
        this.controle = new CalcControle(this.interface_, this.dados);
    }

    public void funcionar() {
        this.interface_.mostraBoasVindas();
        this.dados.reset();
        this.controle.executar();
    }
}

File: Ex-25/ex_25_python/Calculadora.py
from CalcDados import CalcDados
from CalcInterface import CalcInterface
from CalcControle import CalcControle


class Calculadora:
    # cria os componentes e os vínculos (via lista de inicialização)
    def __init__(self):
        self.dados = CalcDados()
        self.interface = CalcInterface()
        self.controle = CalcControle(self.interface, self.dados)

    def funcionar(self):
        self.interface.mostraBoasVindas()
        self.dados.reset()
        self.controle.executar()
